docs/features/deserializing-events.md

```diff
@@ -4,14 +4,28 @@ layout: default
 ---
 
 ## Deserializing Events
-We provide support for deserializing events to typed objects:
+
+We provide support for deserializing events to typed event objects where the custom event payload is available via the `.GetPayload()` method.
+If you want to have the original raw JSON event payload, you can get it via the `.Data` property.
 
 ```csharp
-// From custom event.
-var eventGridBatch = EventGridParser.Parse<NewCarRegistered>(rawEvent);
+// Parse from your custom event implementation with the `.Parse<>` function.
+EventGridBatch<NewCarRegistered> eventGridBatch = EventGridParser.Parse<NewCarRegistered>(rawEvent);
+
+// The event data type will be wrapped inside an `EventGridEvent<>` instance.
+NewCarRegistered eventGridMessage = eventGridBatch.Events.First();
+
+// The original event payload can now be accessed.
+CarEventData typedEventPayload = eventGridMessage.GetPayload();
+object untypedEventPaylaod = eventGridMessage.Data;
+```
+
+Or from event data type objects where the event data payload is available via the `.GetPayload()` method.
+The `.ParseFromData<>` function is especially useful when using directly the Azure services built-in types as shown in the example:
```

Comment:
Same here, newline for the rendering?

![image](https://user-images.githubusercontent.com/4345663/65765526-cca7c800-e128-11e9-8d28-5c4e3fc90f4d.png)

Reply:
This is done, thanks!